sort/pigeonholesort.go

```diff
@@ -19,15 +20,15 @@ func Pigeonhole(arr []int) []int {
 
 	size := max - min + 1
 
-	holes := make([]int, size)
-
+	holes := make([]T, size)
+	// To maintain time complexity O(n + N), this is the reason for having only Integer constraint instead of Ordered.
```

Comment:
Nice! Helpful comment 😄. Maybe this should go at the documentation level comment on line 13 as a 

```Go
// Pigeonhole sorts a slice using pigeonhole sorting algorithm.
// NOTE: To maintain time complexity O(n + N), this is the reason for having
// only Integer constraint instead of Ordered.
func Pigeonhole[T constraints.Integer](arr []T) []T {
```

Reply:
Thanks, I will update soon